File: src/main/java/org/jetbrains/fortran/FortranFileTypeFactory.java
package org.jetbrains.fortran;

import com.intellij.openapi.fileTypes.FileTypeConsumer;
import com.intellij.openapi.fileTypes.FileTypeFactory;
import org.jetbrains.annotations.NotNull;

public class FortranFileTypeFactory extends FileTypeFactory {
    @Override
    public void createFileTypes(@NotNull FileTypeConsumer fileTypeConsumer) {
        for (int i = 0; i < FortranFileType.DEFAULT_EXTENSIONS.length; i++)
        {
            fileTypeConsumer.consume(FortranFileType.INSTANCE, FortranFileType.DEFAULT_EXTENSIONS[i]);
        }
    }
}

File: src/main/java/org/jetbrains/fortran/FortranFileTypeFactory.kt
package org.jetbrains.fortran

import com.intellij.openapi.fileTypes.FileTypeConsumer
import com.intellij.openapi.fileTypes.FileTypeFactory

class FortranFileTypeFactory : FileTypeFactory() {
    override fun createFileTypes(fileTypeConsumer: FileTypeConsumer) {
        for (i in FortranFileType.DEFAULT_EXTENSIONS.indices) {
            fileTypeConsumer.consume(FortranFileType, FortranFileType.DEFAULT_EXTENSIONS[i])
        }
    }
}
